realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt

```diff
@@ -49,6 +44,9 @@ class RealmProxyMediatorGenerator(private val processingEnvironment: ProcessingE
             simpleModelClassNames.add(metadata.simpleJavaClassName)
             internalClassNames.add(metadata.internalClassName)
             embeddedClass.add(metadata.embedded)
+            metadata.primaryKey?.let {
```

Comment:
Nitpick: you aren't using `metadata.primaryKey` in the lambda here, so instead of `let` I would use a good old null check.